pkg/cmd/roachtest/ledger.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+
+package main
+
+import (
+	"context"
+	"fmt"
+)
+
+func registerLedger(r *registry) {
+	const nodes = 6
+	const azs = "us-central1-a,us-central1-b,us-central1-c"
+	r.Add(testSpec{
+		Name:    fmt.Sprintf("ledger/nodes=%d/multi-az", nodes),
+		Cluster: makeClusterSpec(nodes+1, cpu(16), geo(), zones(azs)),
+		Run: func(ctx context.Context, t *test, c *cluster) {
+			// TODO DURING REVIEW: How should we handle the round-robin
```

Comment:
It seems to me like the cases are that either you want 1 workload node total and you want it to be in the first region or you want N workload nodes per region. Both of these are handled in a pretty straightforward way if the zones are distributed in a round-robin fashion rather than the current chunked fashion. That being said, there's still the need for some logic to get the addresses of the nodes for a given region (the `gatewayNodes` here)

Here's my suggestion, let's change the roachprod code to allocate regions in a round-robin fashion (see https://github.com/cockroachdb/cockroach/pull/37492) and then let's add a function to return the nodeListOption for a given zone given a number of zones:

That leaves the detail of migrating existing tests which assume the current behavior which I haven't looked in to.

Reply:
Discussed in person. The solution in #37492 seems like the best of the alternatives.